src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/DotNetCLI.cs

```diff
@@ -69,22 +91,27 @@ public bool Build (string projectOrSolution, string configuration, string target
 			if (string.IsNullOrEmpty (BuildLogFile))
 				BuildLogFile = Path.Combine (testDir, "build.log");
 
-			var execArgs = new string[] {
-				"build", $"\"{projectOrSolution}\"", $"/p:Configuration={configuration}",
-				string.IsNullOrEmpty (target) ? string.Empty : $"/t:{target}",
-				Directory.Exists (AndroidSdkPath) ? $"/p:AndroidSdkDirectory=\"{AndroidSdkPath}\"" : string.Empty,
-				Directory.Exists (AndroidNdkPath) ? $"/p:AndroidNdkDirectory=\"{AndroidNdkPath}\"" : string.Empty,
-				"/noconsolelogger", $"/flp1:LogFile=\"{BuildLogFile}\";Encoding=UTF-8;Verbosity={Verbosity}",
+			var arguments = new List<string> {
+				verb,
+				$"\"{projectOrSolution}\"",
+				$"/p:Configuration={project.Configuration}",
+				"/noconsolelogger",
+				$"/flp1:LogFile=\"{BuildLogFile}\";Encoding=UTF-8;Verbosity={Verbosity}",
 				$"/bl:\"{Path.Combine (testDir, "msbuild.binlog")}\""
 			};
-
-			bool succeeded = Execute (execArgs);
-			if (succeeded && Directory.Exists (testDir)) {
-				FileSystemUtils.SetDirectoryWriteable (testDir);
-				Directory.Delete (testDir, true);
+			if (restore) {
+				arguments.Add ("/restore");
```

Comment:
I don't think we need a restore param here, as all `dotnet` commands should run a `restore` first by default if needed?

Reply:
I think I added this in confusion when something wasn't working 👍